### PR TITLE
Mailerlite - new actions

### LINKED ISSUE #1523: [ACTION] Mailerlite new actions
**Is there a specific app this action is for?**
Mailerlite

**Please provide a link to the relevant API docs for the specific service / operation.**
https://developers.mailerlite.com/docs

**Create Subscriber** \| Create a new subscriber
**Update Subscriber** \| Updates single subscriber by ID or email
**Remove Subscriber from Group** \| Removes single subscriber from specified group
**List Subscribers** \| Lists subscribers by group

### PR DESCRIPTION
**Pull request by michelle0927**
**Create Subscriber** | Create a new subscriber
**Update Subscriber** | Updates single subscriber by ID or email
**Remove Subscriber** from Group | Removes single subscriber from specified group
**List Subscribers** | Lists subscribers by group

Resolves #1523

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/mailerlite/mailerlite.app.mjs [hunk 1]
```diff
@@ -1,11 +1,125 @@
...
+    },
+    async listFields() {
+      const client = await this._getClient();
+      return client.getFields();
+    },
+    async listCampaigns(status = "sent") {
+      const client = await this._getClient();
+      return client.getCampaigns(status);
+    },
+    async createSubscriber(data) { console.log(data);
```
Comment (andrewjschuang) [inline components/mailerlite/mailerlite.app.mjs:108]:
```suggestion
    async createSubscriber(data) {
```

**Review by andrewjschuang — APPROVED** [review]
@michelle0927 thanks for the changes! LGTM!

**Comment by DilanAthukorala** [comment]
@michelle0927 For  unsubscribed, active

**Comment by DilanAthukorala** [comment]
@michelle0927 for update_subscriber action the "unconfirmed" subscriber type is not valid. 
Only valid types are - unsubscribed, active

image

**Comment by michelle0927** [comment]
@DilanAthukorala I made an update to remove "unconfirmed" as an available type in `update-subscriber`.

**Comment by DilanAthukorala** [comment]
@michelle0927 for all actions subscribers prop list only **active** subscribers, Can you please confirm whether this is the expected use case?

**Comment by michelle0927** [comment]
> @michelle0927 for all actions subscribers prop list only **active** subscribers, Can you please confirm whether this is the expected use case?

@DilanAthukorala That's what the API returns. I could add another `type` prop so that the user can select which type of subscriber to return using this endpoint:
https://developers.mailerlite.com/reference/subscribers
and this one for subscribers within groups:
https://developers.mailerlite.com/reference/subscribers-in-a-group-by-type

What do you think?

**Comment by DilanAthukorala** [comment]
> > @michelle0927 for all actions subscribers prop list only **active** subscribers, Can you please confirm whether this is the expected use case?
> 
> @DilanAthukorala That's what the API returns. I could add another `type` prop so that the user can select which type of subscriber to return using this endpoint: https://developers.mailerlite.com/reference/subscribers and this one for subscribers within groups: https://developers.mailerlite.com/reference/subscribers-in-a-group-by-type
> 
> What do you think?

@michelle0927 Yes I think that it would be good to add 'type' prop since some actions (ex - update_subscriber) can be performed on other type of subscribers.

**Comment by michelle0927** [comment]
@DilanAthukorala Despite the documentation, I found that [this endpoint](https://developers.mailerlite.com/reference/subscribers) for retrieving subscribers only returns active subscribers. [The endpoint to get group subscribers](https://developers.mailerlite.com/reference/subscribers-in-a-group-by-type) does retrieve subscribers by type, so I added the ability to select subscriber type if a group has already been selected.

**Comment by DilanAthukorala** [comment]
@michelle0927 Thanks.

Verification completed.

**Pull request merged**